.github/workflows/infracost.yml

```diff
@@ -0,0 +1,74 @@
+name: Infracost CI
+
+on:
+  pull_request:
+    paths:
+      - 'qhub/template/stages/02-infrastructure'
+      - '.github/workflows/infracost.yml'
+
+jobs:
+  infracost:
+    name: Run Infracost on QHub Infrastructure
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: true
+      matrix:
+        provider:
+          - aws
+          - gcp
+          - azure
+
+    env:
+      TF_ROOT: qhub/template/stages/02-infrastructure/${{ matrix.provider }}
```

Comment:
There are other resources that we might want to account for:

https://github.com/Quansight/qhub/blob/main/qhub/template/stages/01-terraform-state/aws/modules/terraform-state/main.tf

Reply:
Added it below to account for all infra updates.